src/jit/lclvars.cpp

```diff
@@ -1917,48 +1887,204 @@ bool Compiler::lvaShouldPromoteStructVar(unsigned lclNum, lvaStructPromotionInfo
     return shouldPromote;
 }
 
-/*****************************************************************************
- * Promote a struct type local */
+//--------------------------------------------------------------------------------------------
+// SortStructFields - sort the fields according to the increasing order of the field offset.
+//
+// Notes:
+//   This is needed because the fields need to be pushed on stack (when referenced as a struct) in the order.
```

Comment:
Nit: I would say "in offset order" or something ("in the order" seems unnecessarily vague)